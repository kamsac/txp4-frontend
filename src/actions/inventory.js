import dotenvConfiguration from '../dotenv-configuration';
import InventoryResourceMock from '../resources/inventory/mock';
import InventoryResource from '../resources/inventory';

export const REQUEST_ITEMS = 'inventory:requestItems';
export function requestItems() {
  return dispatch => {
    dispatch(fetchItems());
  }
}

function fetchItems() {
  const resource = dotenvConfiguration.API_URL ? InventoryResource : InventoryResourceMock;

  return dispatch => {
    resource.getItems()
      .then((response) => {
        dispatch(receiveItems(response.data));
      });
  }
}

export const RECEIVE_ITEMS = 'inventory:receiveItems';
export function receiveItems(items) {
  return {
    type: RECEIVE_ITEMS,
    payload: { items }
  }
}

export const SELECT_ITEM = 'inventory:selectItem';
export function selectItem(selectedItem) {
  return {
    type: SELECT_ITEM,
    payload: { selectedItem }
  }
}

export const EQUIP_ITEM = 'inventory:equipItem';
export function equipItem(item) {
  return {
    type: EQUIP_ITEM,
    payload: { item }
  }
}
