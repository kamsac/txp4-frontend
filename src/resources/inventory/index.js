import axios from 'axios';
import dotenvConfiguration from '../../dotenv-configuration';

const InventoryResource = {
  getItems(userLogin) {
    return axios.get(`${dotenvConfiguration.API_URL}/${userLogin}/items`);
  },

  equipItem(item) {
    return axios.get(`${dotenvConfiguration.API_URL}/`);
  },
};

export default InventoryResource;
