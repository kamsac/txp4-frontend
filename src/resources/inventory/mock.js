const ITEMS = [
  {
    id: "a1a1a1",
    name: "Tire",
    vendorId: 'aaa',
    equipRegion: "tire",
    tier: 3,
    isEquipped: true,
    modifiers: [
      {
        name: "maxSpeed",
        value: 1,
      },
      {
        name: "acceleration",
        value: 1.2,
      },
      {
        name: "steering",
        value: 1.1,
      },
      {
        name: "breaking",
        value: 1.2,
      },
      {
        name: "grip",
        value: 1.5,
      },
      {
        name: "gravity",
        value: 1,
      },
    ],
  },
  {
    id: "b2b2b2",
    name: "Less tired",
    vendorId: 'bbb',
    equipRegion: "tire",
    tier: 1,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 0.4,
      },
      {
        name: "acceleration",
        value: 1,
      },
      {
        name: "steering",
        value: 1.4,
      },
      {
        name: "breaking",
        value: 2,
      },
      {
        name: "grip",
        value: 3,
      },
      {
        name: "gravity",
        value: 1,
      },
    ],
  },
  {
    id: "c3c3c3",
    name: "Shit engine",
    vendorId: 'ccc',
    equipRegion: "engine",
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 2,
      },
      {
        name: "acceleration",
        value: 1.8,
      },
      {
        name: "steering",
        value: 1,
      },
      {
        name: "breaking",
        value: 1,
      },
      {
        name: "grip",
        value: 0.2,
      },
      {
        name: "gravity",
        value: 1.5,
      },
    ],
  },
  {
    id: "d4d4d4",
    name: "Some tire",
    vendorId: 'ddd',
    equipRegion: "tire",
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 1,
      },
      {
        name: "acceleration",
        value: 1,
      },
      {
        name: "steering",
        value: 1,
      },
      {
        name: "breaking",
        value: 1.1,
      },
      {
        name: "grip",
        value: 0.8,
      },
      {
        name: "gravity",
        value: 0.9,
      },
    ],
  },
  {
    id: "e5e5e5",
    name: "Tire",
    vendorId: 'aaa',
    equipRegion: "tire",
    tier: 3,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 1,
      },
      {
        name: "acceleration",
        value: 1.2,
      },
      {
        name: "steering",
        value: 1.1,
      },
      {
        name: "breaking",
        value: 1.2,
      },
      {
        name: "grip",
        value: 1.5,
      },
      {
        name: "gravity",
        value: 1,
      },
    ],
  },
  {
    id: "f6f6f6",
    name: "Duper engine",
    vendorId: 'aaa',
    equipRegion: "engine",
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 1.5,
      },
      {
        name: "acceleration",
        value: 3,
      },
      {
        name: "steering",
        value: 1,
      },
      {
        name: "breaking",
        value: 1,
      },
      {
        name: "grip",
        value: 1,
      },
      {
        name: "gravity",
        value: 1.5,
      },
    ],
  },
  {
    id: "f7f7f7",
    name: "Tire",
    vendorId: 'aaa',
    equipRegion: "tire",
    tier: 3,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 1,
      },
      {
        name: "acceleration",
        value: 1.2,
      },
      {
        name: "steering",
        value: 1.1,
      },
      {
        name: "breaking",
        value: 1.2,
      },
      {
        name: "grip",
        value: 1.5,
      },
      {
        name: "gravity",
        value: 1,
      },
    ],
  },
  {
    id: "g8g8g8",
    name: "Less tired",
    vendorId: 'bbb',
    equipRegion: "tire",
    tier: 1,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 0.4,
      },
      {
        name: "acceleration",
        value: 1,
      },
      {
        name: "steering",
        value: 1.4,
      },
      {
        name: "breaking",
        value: 2,
      },
      {
        name: "grip",
        value: 3,
      },
      {
        name: "gravity",
        value: 1,
      },
    ],
  },
  {
    id: "h9h9h9",
    name: "Shit engine",
    vendorId: 'ccc',
    equipRegion: "engine",
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 2,
      },
      {
        name: "acceleration",
        value: 1.8,
      },
      {
        name: "steering",
        value: 1,
      },
      {
        name: "breaking",
        value: 1,
      },
      {
        name: "grip",
        value: 0.2,
      },
      {
        name: "gravity",
        value: 1.5,
      },
    ],
  },
  {
    id: "i0i0i0",
    name: "Some tire",
    vendorId: 'ddd',
    equipRegion: "tire",
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 1,
      },
      {
        name: "acceleration",
        value: 1,
      },
      {
        name: "steering",
        value: 1,
      },
      {
        name: "breaking",
        value: 1.1,
      },
      {
        name: "grip",
        value: 0.8,
      },
      {
        name: "gravity",
        value: 0.9,
      },
    ],
  },
  {
    id: "j1j1j1",
    name: "Transmission",
    vendorId: 'ccc',
    equipRegion: "transmission",
    tier: 3,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 0.7,
      },
      {
        name: "acceleration",
        value: 1.5,
      },
      {
        name: "steering",
        value: 1,
      },
      {
        name: "breaking",
        value: 1,
      },
      {
        name: "grip",
        value: 1,
      },
      {
        name: "gravity",
        value: 1,
      },
    ],
  },
];

const InventoryResourceMock = {
  getItems() {
    return new Promise((resolve, reject) => {
      const response = {
        data: ITEMS,
      };
      setTimeout(() => resolve(response), 100);
    });
  },

  equipItem(item) {
    return new Promise((resolve, reject) => {
      const response = {
        data: ITEMS,
      };
      setTimeout(() => resolve(ITEMS), 100);
    });
  },
};

export default InventoryResourceMock;
