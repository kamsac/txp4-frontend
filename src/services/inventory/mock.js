const ITEMS = [
  {
    name: "Tire",
    vendor: {
      name: "Super",
      color: "#8f0",
    },
    type: "tire",
    tier: 3,
    modifiers: [
      {
        name: "maxSpeed",
        value: 1,
      },
      {
        name: "acceleration",
        value: 1.2,
      },
      {
        name: "steering",
        value: 1.1,
      },
      {
        name: "breaking",
        value: 1.2,
      },
      {
        name: "grip",
        value: 1.5,
      },
      {
        name: "gravity",
        value: 1,
      },
    ],
  },
  {
    name: "Less tired",
    vendor: {
      name: "Safe",
      color: "#f80",
    },
    type: "tire",
    tier: 1,
    modifiers: [
      {
        name: "maxSpeed",
        value: 0.4,
      },
      {
        name: "acceleration",
        value: 1,
      },
      {
        name: "steering",
        value: 1.4,
      },
      {
        name: "breaking",
        value: 2,
      },
      {
        name: "grip",
        value: 3,
      },
      {
        name: "gravity",
        value: 1,
      },
    ],
  },
  {
    name: "Shit engine",
    vendor: {
      name: "Heavy",
      color: "#800",
    },
    type: "engine",
    tier: 2,
    modifiers: [
      {
        name: "maxSpeed",
        value: 2,
      },
      {
        name: "acceleration",
        value: 1.8,
      },
      {
        name: "steering",
        value: 1,
      },
      {
        name: "breaking",
        value: 1,
      },
      {
        name: "grip",
        value: 0.2,
      },
      {
        name: "gravity",
        value: 1.5,
      },
    ],
  },
  {
    name: "Some tire",
    vendor: {
      name: "Light",
      color: "#08f",
    },
    type: "tire",
    tier: 2,
    modifiers: [
      {
        name: "maxSpeed",
        value: 1,
      },
      {
        name: "acceleration",
        value: 1,
      },
      {
        name: "steering",
        value: 1,
      },
      {
        name: "breaking",
        value: 1.1,
      },
      {
        name: "grip",
        value: 0.8,
      },
      {
        name: "gravity",
        value: 0.9,
      },
    ],
  },
  {
    name: "Tire",
    vendor: {
      name: "Super",
      color: "#8f0",
    },
    type: "tire",
    tier: 3,
    modifiers: [
      {
        name: "maxSpeed",
        value: 1,
      },
      {
        name: "acceleration",
        value: 1.2,
      },
      {
        name: "steering",
        value: 1.1,
      },
      {
        name: "breaking",
        value: 1.2,
      },
      {
        name: "grip",
        value: 1.5,
      },
      {
        name: "gravity",
        value: 1,
      },
    ],
  },
  {
    name: "Duper engine",
    vendor: {
      name: "Super",
      color: "#8f0",
    },
    type: "engine",
    tier: 2,
    modifiers: [
      {
        name: "maxSpeed",
        value: 1.5,
      },
      {
        name: "acceleration",
        value: 3,
      },
      {
        name: "steering",
        value: 1,
      },
      {
        name: "breaking",
        value: 1,
      },
      {
        name: "grip",
        value: 1,
      },
      {
        name: "gravity",
        value: 1.5,
      },
    ],
  },
];

const InventoryService = {
  getItems(cb) {
    setTimeout(() => cb(ITEMS), 100);
  },

  equipItem(item, cb) {
    setTimeout(() => cb(item), 100);
  }
};

export default InventoryService;