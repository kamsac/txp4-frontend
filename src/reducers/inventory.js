import {EQUIP_ITEM, RECEIVE_INVENTORY} from '../actions/inventory';

const initialState = {
  items: [],
};

export function inventoryReducer(state = initialState, action) {
  switch(action.type) {
    case RECEIVE_INVENTORY:
      return Object.assign({}, state, {
        items: action.payload.items,
      });

    case EQUIP_ITEM:
      const interpolatedItems = [...state.items]
        .map((inventoryItem) => {
          if (inventoryItem.equipRegion === action.payload.item.equipRegion) {
            inventoryItem.isEquipped = false;
          }

          if (inventoryItem.id === action.payload.item.id) {
            inventoryItem.isEquipped = true;
          }

          return inventoryItem;
        });

      return Object.assign({}, state, {
        items: interpolatedItems,
      });
    default:
      return state;
  }
}
