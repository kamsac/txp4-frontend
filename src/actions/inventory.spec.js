import { equipItem, receiveItems } from './inventory';

test('receiveItems', () => {
  const someItems = [
    {
      id: 'a1a1a1',
      name: 'Tire',
      vendorId: 'aaa',
      equipRegion: 'tire',
      tier: 3,
      isEquipped: true,
      modifiers: [
        {
          name: 'acceleration',
          value: 1.2,
        },
      ],
    },
    {
      id: 'b2b2b2',
      name: 'Less tired',
      vendorId: 'bbb',
      equipRegion: 'tire',
      tier: 1,
      isEquipped: false,
      modifiers: [
        {
          name: 'maxSpeed',
          value: 0.4,
        },
      ],
    },
  ];

  expect(receiveItems(someItems)).toMatchSnapshot();
});

test('equipItem', () => {
  const someItem = {
    id: 'e5e5e5',
    name: 'Tire',
    vendorId: 'aaa',
    equipRegion: 'tire',
    tier: 3,
    isEquipped: false,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 1,
      },
    ],
  };

  expect(equipItem(someItem)).toMatchSnapshot();
});
