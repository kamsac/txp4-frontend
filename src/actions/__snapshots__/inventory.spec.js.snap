// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`equipItem 1`] = `
Object {
  "payload": Object {
    "item": Object {
      "equipRegion": "tire",
      "id": "e5e5e5",
      "isEquipped": false,
      "modifiers": Array [
        Object {
          "name": "maxSpeed",
          "value": 1,
        },
      ],
      "name": "Tire",
      "tier": 3,
      "vendorId": "aaa",
    },
  },
  "type": "inventory:equipItem",
}
`;

exports[`receiveItems 1`] = `
Object {
  "payload": Object {
    "items": Array [
      Object {
        "equipRegion": "tire",
        "id": "a1a1a1",
        "isEquipped": true,
        "modifiers": Array [
          Object {
            "name": "acceleration",
            "value": 1.2,
          },
        ],
        "name": "Tire",
        "tier": 3,
        "vendorId": "aaa",
      },
      Object {
        "equipRegion": "tire",
        "id": "b2b2b2",
        "isEquipped": false,
        "modifiers": Array [
          Object {
            "name": "maxSpeed",
            "value": 0.4,
          },
        ],
        "name": "Less tired",
        "tier": 1,
        "vendorId": "bbb",
      },
    ],
  },
  "type": "inventory:receiveItems",
}
`;

exports[`selectItem 1`] = `
Object {
  "payload": Object {
    "selectedItem": Object {
      "equipRegion": "tire",
      "id": "e5e5e5",
      "isEquipped": false,
      "modifiers": Array [
        Object {
          "name": "maxSpeed",
          "value": 1,
        },
      ],
      "name": "Tire",
      "tier": 3,
      "vendorId": "aaa",
    },
  },
  "type": "inventory:selectItem",
}
`;
